### PR TITLE
React Hot Loader

### LINKED ISSUE #15: Job pages aren't templating the input names correctly
Right now the field names are `in-{ name }`, `param-{ name }`, `out-{ name }`. These end up masking each other, we just need to fix this so that `name` is interpolated into the string correctly.

### LINKED ISSUE #17: Create 'Cancel' button on job pages
In order to support backwards navigation in Electron, a cancel button would allow the user to travel back to the main page without having to execute the job.

### PR DESCRIPTION
**Pull request by jakereps**
This accidentally fixes #15 and fixes #17. Copied code over from the other PR to containerize the job pages that had those fixes in it.

### CODE REVIEW

File: app/js/components/pages/Flow.jsx [hunk 1]
```diff
@@ -56,35 +46,44 @@ const Flow = ({ routeParams: { pluginId, flowId } }, { store }) => {
...
-
-
-                <button type="submit" className="btn btn-primary">Go!</button>
             </form>
+            <button
+                className="btn btn-primary"
+                onClick={onClickSubmit}
+            >
+                Go!
+            </button>
```
Comment (ebolyen) [inline app/js/components/pages/Flow.jsx]: I would flip the order of these in the UI, `Cancel` should be a "back" operation, whereas `Go` is a move "forward" operation.

**Comment by ebolyen** [comment]
This is ridiculously cool! 

One thing, we need to update ` "react-router": "^2.0.1"` to ` "react-router": "^2.4.0"`.

**Comment by jakereps** [comment]
@ebolyen, Oh yeah! I'll get that updated. Copied the Job page over from the other PR that has it updated (that's why it has the Go/Cancel buttons done too). Forgot that it's only working for me because node modules carry through branches, since they aren't tracked.

**Comment by jakereps** [comment]
Should be good to go, @ebolyen!

**Comment by ebolyen** [comment]
Thanks @jakereps!

**Pull request merged**